Orange/widgets/visualize/owmosaic.py

```diff
@@ -383,19 +382,34 @@ def init_combos(self, data):
             if attr.is_discrete or attr.is_continuous:
                 for combo in self.attr_combos:
                     combo.addItem(icons[attr], attr.name)
+        self.cb_attr_color.addItem("(Pearson residuals)")
+        for attr in chain(data.domain.class_vars, data.domain.attributes, data.domain.metas):
+            if attr.is_discrete or attr.is_continuous:
+                self.cb_attr_color.addItem(icons[attr], attr.name)
 
+    def init_combos(self, data):
+        for combo in self.attr_combos:
+            combo.clear()
+        self.cb_attr_color.clear()
+        if data is None:
+            return
+
+        self.fill_combos(data)
         if self.attr_combos[0].count() > 0:
             self.variable1 = self.attr_combos[0].itemText(0)
             self.variable2 = self.attr_combos[1].itemText(
                 2 * (self.attr_combos[1].count() > 2))
         self.variable3 = self.attr_combos[2].itemText(0)
         self.variable4 = self.attr_combos[3].itemText(0)
+        idx = 1 if data.domain.class_var else 0
+        self.variable_color = self.cb_attr_color.itemText(idx)
 
     def get_attr_list(self):
         return [
             a for a in [self.variable1, self.variable2,
                         self.variable3, self.variable4]
-            if a and a != "(None)"]
+            if a and a != "(None)"], \
+               self.variable_color if self.variable_color != "(Pearson residuals)" else None
```

Comment:
It looks to me like this method is now only called in one additional place (L484), where you discard the first list of attributes and only use the color variable.
Why don't you just use `self.variable_color` there, and you even have an `if` already (just change "is None" to "== Pearson")

I would, however, prefer to put the string "(Pearson residuals)" in a constant at the top and use that. You currently already use the hard-coded string 7 times in this PR.